src/gui/attributetable/qgsattributetableview.h

```diff
@@ -180,6 +180,13 @@ class GUI_EXPORT QgsAttributeTableView : public QgsTableView
      */
     Q_DECL_DEPRECATED void finished() SIP_DEPRECATED;
 
+    /**
+     * Emitted when a row header is double-clicked to request zooming to the feature.
+     * \param fid the feature ID of the double-clicked row
+     * \since QGIS 3.46
+     */
+    void zoomToFeatureRequested( QgsFeatureId fid );
```

Comment:
Let's keep the signal generic:

```suggestion
    /**
     * Emitted when a row header is double-clicked.
     * \param fid the feature ID of the double-clicked row.
     * \since QGIS 4.0
     */
    void rowHeaderDoubleClicked( QgsFeatureId fid );
```